### PR TITLE
[Compute] `az vm update`: Add support of Gen1 VM to trusted launch upgrade

### PR DESCRIPTION
**Pull request by Jing-song**
**Related command**

`az vm update`

**Description**

Gen1 VM to Trusted launch upgrade support
close https://github.com/Azure/azure-cli/issues/29007

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/custom.py [hunk 1]
```diff
@@ -1623,33 +1623,34 @@ def update_vm(cmd, resource_group_name, vm_name, os_disk=None, disk_caching=None
...
+            disk_name = os_disk_id_parsed['name']
 
         if disk_name is not None:
-            from ._vm_utils import validate_update_vm_trusted_launch_supported
-
-            validate_update_vm_trusted_launch_supported(cmd=cmd, vm=vm, os_disk_resource_group=disk_resource_group,
-                                                        os_disk_name=disk_name)
-            # Set --enable-secure-boot False and --enable-vtpm True if not specified by end user.
-            enable_secure_boot = enable_secure_boot if enable_secure_boot is not None else False
+            # Set --enable-secure-boot True and --enable-vtpm True if not specified by end user.
```
Comment (Jing-song) [inline src/azure-cli/azure/cli/command_modules/vm/custom.py:1652]: Request from service team: We should update Gen2 to Trusted launch upgrade logic and set --enable-secure-boot to true by default (if not specific by user).

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py:1652]: Does modifying the default values seem to cause a breaking change in the customer's automated scripts or workflow?

Comment (Jing-song) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py:1652]: It'll not affect workflow or existing script.
Only potential risk is if they've unsigned drivers running on OS, they'll fail to load with OS boot. It is low % risk approved by Azure Security LT as acceptable.

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/custom.py:1652]: Got it, thanks~

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**